dart-impl/mpi/src/dart_communication.c

```diff
@@ -2193,6 +2193,54 @@ dart_ret_t dart_allreduce(
   return DART_OK;
 }
 
+dart_ret_t dart_alltoall(
+    const void *    sendbuf,
+    void *          recvbuf,
+    size_t          nelem,
+    dart_datatype_t dtype,
+    dart_team_t     teamid)
+{
+  DART_LOG_TRACE("dart_alltoall() team:%d nelem:%" PRIu64 "", teamid, nelem);
+
+  CHECK_IS_BASICTYPE(dtype);
+
+  /*
+   * MPI uses offset type int, do not copy more than INT_MAX elements:
+   */
+  if (dart__unlikely(nelem > MAX_CONTIG_ELEMENTS)) {
+    DART_LOG_ERROR("dart_allreduce ! failed: nelem (%zu) > INT_MAX", nelem);
```

Comment:
Can we recover from this? It is also not documented in the interface that there is a limit.